libs/core/src/lib/fixed-card-layout/fixed-card-layout.component.ts

```diff
@@ -0,0 +1,191 @@
+import { CdkDragDrop, CdkDropList, moveItemInArray } from '@angular/cdk/drag-drop';
+import {
+    AfterContentInit,
+    AfterViewChecked,
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    QueryList,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+import { Observable, of, from, fromEvent, Subscription, Subject } from 'rxjs';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+
+const CARD_MINIMUM_WIDTH = 320; // in px; 20rem max card size
+const CARD_GAP_WIDTH = 16; // gap=1rem==16px
+
+@Directive({ selector: '[fdCardDef]' })
+export class CardDefinitionDirective {
+    constructor(public template: TemplateRef<any>) {}
+}
+
+class Layout {
+    constructor(public noOfColumns: number, public screenSize: number) {}
+}
+
+class CardDropped {
+    constructor(public container: CdkDropList, public previousIndex: number, public currentIndex: number) {}
+}
+
+type CardColumn = CardDefinitionDirective[];
+
+@Component({
+    selector: 'fd-fixed-card-layout',
+    templateUrl: './fixed-card-layout.component.html',
+    styleUrls: ['./fixed-card-layout.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FixedCardLayoutComponent implements OnInit, AfterContentInit, AfterViewInit, AfterViewChecked, OnDestroy {
+    /** @hidden */
+    @ContentChildren(CardDefinitionDirective)
+    cards: QueryList<CardDefinitionDirective>;
+
+    /** @hidden */
+    @ViewChild('layout')
+    layout: ElementRef;
+
+    /** Drag drop behaviour can be disabled */
+    @Input()
+    disableDragDrop: boolean;
+
+    /** Event to emit, when layout changes */
+    @Output()
+    layoutChange: EventEmitter<Layout> = new EventEmitter<Layout>();
+
+    /** Event to emit on Card dragged and dropped */
+    @Output()
+    cardDraggedDropped: EventEmitter<CardDropped> = new EventEmitter<CardDropped>();
+
+    /** Array of CardDefinitionDirective Array.To make Table kind of layout.*/
+    public columns: CardColumn[];
+    /** @hidden Number of Columns in layout */
+    private numberOfColumns = 3;
```

Comment:
Why `3` as initial value?
Missing `_` prefix for private properties.

Reply:
fixed.